### ISSUE TITLE
auto-ROI fundamental bugs

### ISSUE THREAD

**Issue by raacampbell** [SainsburyWellcomeCentre/BakingTray#241]
This is an problem dump for general concerns or bugs with the auto-ROI. Once it seems to correctly image a brain and the following issues are dealt with, we close this. 

* [x] May need the image origin (0,0 coordinate) and the stage front/left to be the same thing. Otherwise if the image gets larger during a scan the stage front/left coordinate will no longer mean anything if it expands along the image rows (X for the stage). 9cb33c8e97208f4ac39b5811e6c4edd3327f2489
* [x] Need to shift the tile pattern up by one tile in two locations. Not sure why. See `getNextROIs` (26f0744c3ba014315336ade2df1d4c65efef5d8e, a98e644d6cf33b79647fcec0fd7dc790053290c0, and 07d9026af55df4e1e066a1d26cc277e5adec3a10) and `overlayTileGridOnImage` (d1e20a7ccc98ffe5beab5e667a281712ed09a439). The first is needed for the plotted file grid to match the auto-ROI results. The second to avoid loss of the sample when running `autoROI` -- it slowly eats away the top of the sample. A major reason was that the shifted ROIs weren't being applied: 20f1b2bba3558387bdeff94ba5a1749a50cf5712 But also the dX and dY shifts had to be inverted. 
* [x] The first bake preview is garbled and this messes up `autoROI`. The rest are fine. The preview scan is fine. 
* [x] `autoROI` is failing to expand ROIs at the right edge. Gradually the brain falls out at this end. Possibly this is because we are shifting the ROIs up by one tile each time through the bake loop (See `getNextROIs` -- 26f0744c3ba014315336ade2df1d4c65efef5d8e). YES: that's the problem. If we also add a row of tiles on the bottom then it works (a98e644d6cf33b79647fcec0fd7dc790053290c0) but I have no idea why all this is needed, which is obviously a really bad thing. As above: 20f1b2bba3558387bdeff94ba5a1749a50cf5712

**Comment by raacampbell** [comment]
Something wrong with the bounding boxes drawn during bake. The preview is fine:
![initial](https://user-images.githubusercontent.com/5427811/80985006-89cc7200-8e26-11ea-9887-c0fb5f25ec5a.png)
But the boxes drawn over section 1 are wrong:
![after_first_section](https://user-images.githubusercontent.com/5427811/80985038-95b83400-8e26-11ea-8448-b6e7664aac5b.png)

Severity varies by sample. The above is `SWC037_SWC012_DY006_SWC015_previewStack` and is one of the badly affected ones. Possibly this is an error associated with the front/left position used by the auto-ROI being wrong.

**Comment by raacampbell** [comment]
Image origin and F/L now are in the same place (9cb33c8e97208f4ac39b5811e6c4edd3327f2489) and most things seem fine. The on thing that's wrong is that the ROIs returned by `autoROI` are one tile below where they should be. Looks like they are using the wrong Y origin. So the tiles are being placed by they lower left corder not top-right corner. We'll need to change that and then maybe will need to modify the existing preview images somehow once that's done.

**Comment by raacampbell** [comment]
I have also noticed that the top brain sometimes is clipped. There is something odd here. The shifted BoundingBox positions obtained in `getNextROIs` contain negative numbers. They shouldn't! They should be valid pixels in the current preview image. I tried setting different F/L and image size for the rapid preview and get incorrect bounding boxes:
![Screen Shot 2020-05-15 at 16 24 48](https://user-images.githubusercontent.com/5427811/82067602-daff2000-96c8-11ea-8f8a-08bb7e787797.png)
Maybe figuring out the solution to this will reveal what is going wrong?
